src/api/VitessceConfig.js

```diff
@@ -0,0 +1,401 @@
+import { getNextScope, fromEntries } from '../utils';
+import { COORDINATION_TYPES } from '../app/state/coordination';
+
+
+/**
+ * Class representing a file within a Vitessce config dataset.
+ */
+export class VitessceConfigDatasetFile {
+  /**
+   * Construct a new file definition instance.
+   * @param {string} url The URL to the file.
+   * @param {string} dataType The type of data contained in the file.
+   * @param {string} fileType The file type.
+   */
+  constructor(url, dataType, fileType) {
+    this.file = {
+      url,
+      type: dataType,
+      fileType,
+    };
+  }
+
+  /**
+   * @returns {object} This dataset file as a JSON object.
+   */
+  toJSON() {
+    return this.file;
+  }
+}
+
+/**
+ * Class representing a dataset within a Vitessce config.
+ */
+export class VitessceConfigDataset {
+  /**
+   * Construct a new dataset definition instance.
+   * @param {string} uid The unique ID for the dataset.
+   * @param {string} name The name of the dataset.
+   */
+  constructor(uid, name) {
+    this.dataset = {
+      uid,
+      name,
+      files: [],
+    };
+  }
+
+  /**
+   * Add a file definition to the dataset.
+   * @param {string} url The URL to the file.
+   * @param {string} dataType The type of data contained in the file.
+   * @param {string} fileType The file type.
+   * @returns {VitessceConfigDataset} This, to allow chaining.
+   */
+  addFile(url, dataType, fileType) {
+    this.dataset.files.push(new VitessceConfigDatasetFile(url, dataType, fileType));
+    return this;
+  }
+
+  /**
+   * @returns {object} This dataset as a JSON object.
+   */
+  toJSON() {
+    return {
+      ...this.dataset,
+      files: this.dataset.files.map(f => f.toJSON()),
+    };
+  }
+}
+
+/**
+ * Class representing a view within a Vitessce layout.
+ */
+export class VitessceConfigView {
+  /**
+   * Construct a new view instance.
+   * @param {string} component The name of the Vitessce component type.
+   * @param {object} coordinationScopes A mapping from coordination type
+   * names to coordination scope names.
+   * @param {number} x The x-coordinate of the view in the layout.
+   * @param {number} y The y-coordinate of the view in the layout.
+   * @param {number} w The width of the view in the layout.
+   * @param {number} h The height of the view in the layout.
+   */
+  constructor(component, coordinationScopes, x, y, w, h) {
+    this.view = {
```

Comment:
Why not just set these as attributes?  It seems redundant to have a `.view` when the class is also called "View" (same for other classes)

Reply:
That is true. Although I think I just did it this way so that there is some compartmentalization, so that it is easy to implement the `.toJSON` function. For example, there could be future point when we need to add an internal variable, which we do not want to be included in the "serialized" output, and we could just add it onto `this` without thinking about how it affects the `.toJSON` return value.

Reply:
Ok sounds good.  Good to merge whenever.